import_export/resources.py

```diff
@@ -609,18 +609,17 @@ def skip_row(self, instance, original, row, import_validation_errors=None):
         if not self._meta.skip_unchanged or self._meta.skip_diff or import_validation_errors:
             return False
         for field in self.get_import_fields():
-            try:
-                # For fields that are models.fields.related.ManyRelatedManager
-                # we need to compare the results
-                if isinstance(field.widget, widgets.ManyToManyWidget):
-                    # compare with the future value to detect changes
-                    instance_value = list(field.clean(row))
-                else:
-                    instance_value = list(field.get_value(instance).all())
+            # For fields that are models.fields.related.ManyRelatedManager
+            # we need to compare the results
+            if isinstance(field.widget, widgets.ManyToManyWidget):
+                instance_values = list(field.clean(row))
+                original_values = list(field.get_value(original).all())
+                if len(instance_values) != len(original_values):
+                    return False
 
-                if instance_value != list(field.get_value(original).all()):
+                if sorted([v.id for v in instance_values]) != sorted([v.id for v in original_values]):
```

Comment:
I can’t remember for sure but I don’t think you need to make a list out of this, I think you can do generators and sorted will spit out a list, and it winds out being maybe a hair more performant.

Reply:
not sure what you mean here - please can you elaborate?

Reply:
believe you can do 

` sorted(v.id for v in instance_values) instead of if sorted ([v.id for v in instance_values]) `

and it should be modestly more performant with large datasets maybe (since I think sorted returns a new list, you don't make two lists).

Reply:
good call.